src/connection.cpp

```diff
@@ -151,8 +140,41 @@ void Connection::rpc(const std::string& function, ManagementResponseCallback res
         throw std::invalid_argument(
             "trying to register management RPC response callback that is already registered");
     }
+}
+
+std::string Connection::prepare_message(const std::string& function, json payload)
+{
+    payload["function"] = function;
+    return payload.dump();
+}
+
+std::unique_ptr<AMQP::Envelope> Connection::prepare_rpc_envelop(const std::string& message)
```

Comment:
That `AMQP::Envelope` is not even movable? Surely seems like a issue/PR to me. But yeah, keep the `unique_ptr<Envelope>` for now.

Reply:
https://github.com/CopernicaMarketingSoftware/AMQP-CPP/commit/00b81949d3736470362e9f5b5324b8e1973867bb